src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1417,12 +1417,41 @@ public bool Equals(ModuleHandle handle)
 
         public static bool operator !=(ModuleHandle left, ModuleHandle right) => !left.Equals(right);
 
-        #region Internal FCalls
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern IRuntimeMethodInfo GetDynamicMethod(Reflection.Emit.DynamicMethod method, RuntimeModule module, string name, byte[] sig, Resolver resolver);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ModuleHandle_GetDynamicMethod", StringMarshalling = StringMarshalling.Utf8)]
+        private static partial void GetDynamicMethod(
+            QCallModule module,
+            string name,
+            byte* sig,
+            int sigLen,
+            ObjectHandleOnStack resolver,
+            ObjectHandleOnStack result);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern int GetToken(RuntimeModule module);
+        internal static IRuntimeMethodInfo GetDynamicMethod(Reflection.Emit.DynamicMethod method, RuntimeModule module, string name, byte[] sig, Resolver resolver)
+        {
+            if (module is null)
+            {
+                throw new ArgumentNullException(SR.Arg_InvalidHandle);
+            }
+
```

Comment:
```suggestion
```
I do not think we need this. If there is something horibly wrong and we end up with `module == null` here, we are going to get NullReferenceException in `new QCallModule(ref module)` that is about as hard to diagnose as this explicit exception.